src/cycles/incremental.jl

```diff
@@ -0,0 +1,266 @@
+using Base.Iterators: repeated
+
+# Abstract Interface
+
+"""
+    abstract type IncrementalCycleTracker
+
+The supertype for incremental cycle detection problems. The abstract type
+constructor IncrementalCycleTracker(G) may be used to automatically select
+a specific incremental cycle detection algorithm. See [`add_edge_checked!`](dfsdfsdfsdfref)
+for a usage example.
+"""
+abstract type IncrementalCycleTracker{I} <: AbstractGraph{I} end
+
+function (::Type{IncrementalCycleTracker})(s::AbstractGraph{I}; dir::Symbol=nothing) where {I}
```

Comment:
We should change `::Symbol` to `::Union{nothing, Symbol}`

Reply:
@etiennedeg thanks! fixed.